examples/bri-3/src/bri/identity/bpiSubjects/agents/bpiSubjects.agent.ts

```diff
@@ -14,7 +14,7 @@ export class BpiSubjectAgent {
     };
   } 
 
-  public createNewExternalBpiSubject(name :string, desc: string, pk: string): BpiSubject {
-    return new BpiSubject(name, desc, BpiSubjectType.External, pk);
+  public createNewExternalBpiSubject(name :string, description: string, publicKey: string): BpiSubject {
+    return new BpiSubject(<BpiSubject>{ name, description, type: BpiSubjectType.External, publicKey });
```

Comment:
Seems redundant to add a typed object as a param to a ctor of the class of the same object. It is the ctor that defines how is the object of that type constructed, so plain params are completely ok in my eyes.

Reply:
No actually this is to add type safety so that constructor accepts only the allowed types

Reply:
Constructor already has type safe params since we are explicitily listing the parameters with their types (name: string, description: string..) You will not be able to pass in any other type to the ctor but those params with their defined types. For me this approach is redundant, not sure what others think.

Reply:
Okay, if you think this is redundant, could you suggest a better approach? 
Also, if you can suggest a better approach considering we want to initialize the class object with only 1 field out of 3, for example?

Reply:
@Manik-Jain something like this could work, no?

```
class test {    
 constructor(inputName: string, inputAge: number, address ? : string) {    
  this.name = inputName;    
  this.age = inputAge;    
  this.Address = address;    
 }    
 name: string;    
 age: number;    
 IsFemale: boolean;    
 Address: string | undefined;    
} 
```

Reply:
yeah the change that I've done is a little cosmetic on this itself.

In the previous approach, the invocation will be:

const test:Test = new Test('XYZ', 20);
const test2:Test = new Test('XYZ', 20, true, 'some address');

```
class test {    
 constructor(init: Test) {    
  Object.assign(this, init);  
 }    
 name: string;    
 age: number;    
 IsFemale?: boolean;    
 Address?: string | undefined;    
} ```

the invocation will be purely object based:

for the 2 mandate variables:
```
const test:Test = new Test(< Test > {
name : 'XYZ',
age : 23
})
```

for all variables:

```
const test:Test = new Test(< Test > {
name : 'XYZ',
age : 23,
IsFemale: true;    
 Address: 'some Address';    
})
```

The major difference is better use of object notation, and reduction in number of code lines inside the constructor.

Reply:
> yeah the change that I've done is a little cosmetic on this itself.
> 
> In the previous approach, the invocation will be:
> 
> ```
> const test:Test = new Test('XYZ', 20);
> const test2:Test = new Test('XYZ', 20, true, 'some address');
> ```
> 
> With the proposed approach, there will be a considerable reduction in the constructor code,
> 
> ```
> class test {    
>  constructor(init: Test) {    
>   Object.assign(this, init);  
>  }    
>  name: string;    
>  age: number;    
>  IsFemale?: boolean;    
>  Address?: string | undefined;    
> } 
> ```
> 
> the invocation will be purely object based:
> 
> for the 2 mandate variables:
> 
> ```
> const test:Test = new Test(< Test > {
> name : 'XYZ',
> age : 23
> })
> ```
> 
> for all variables:
> 
> ```
> const test:Test = new Test(< Test > {
> name : 'XYZ',
> age : 23,
> IsFemale: true;    
>  Address: 'some Address';    
> })
> ```
> 
> The major difference is better use of object notation, and reduction in number of code lines inside the constructor.

@Manik-Jain did u just give me a complement by saying that what I suggest makes sense? ;-)

Reply:
change made.